### PR TITLE
Blaze: show an error if one of the requirements of the creation is missing

### LINKED ISSUE #10834: Blaze: validate campaign requirements before starting the creation
The campaign creation requires:
1. Selecting an image.
2. Entering the ad's content (title and description)

If one of the two is missing, we should let the user know, and guide them to fix the issue.

### PR DESCRIPTION
**Pull request by hichamboushaba**
Closes: #10834 

### Description
Blaze campaigns require an image and some content, currently we don't handle validating the presence of both of them, this PR adds this, and guides the user to add the missing parts.

The Ad's tagline and description are fetched from the API, but if the call fails, they'll stay empty, so we still need to handle this.

This PR's handling of the image is aligned with what iOS did here https://github.com/woocommerce/woocommerce-ios/pull/11771

##### Image
1. Create a product that doesn't have an image.
2. Open blaze campaign creation for this product.
3. Click on Confirm Details.
4. Notice the dialog.
5. Click on Add image.
6. Confirm this opens the Ad details screen.

##### Tagline and/or description
1. Disable internet connection.
2. Open blaze campaign creation form.
3. Click on Confirm Details.
4. Notice the dialog.
5. Click on "Add"
6. Confirm this opens the Ad details screen.

### Images/gif
| Image missing | tagline or description missing |
| --- | --- |
| Screenshot_20240215_180012 | Screenshot_20240215_175937 |

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          3 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><code>strings.xml</code> files should only be updated on release branches, when the translations are downloaded by our automation.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Class <code>DialogState</code> is missing tests, but <code>unit-tests-exemption</code> label was set to ignore this.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-android/milestone/193">17.4</a>. The due date for this milestone has already passed.<br />
Please assign it to a milestone with a later deadline or check whether the release for this milestone has already been finished.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -3843,6 +3843,10 @@
     <string name="blaze_campaign_preview_days_duration">%1$s days from %2$s</string>
     <string name="blaze_campaign_preview_target_default_value">All</string>
     <string name="blaze_campaign_preview_target_location_search_hint">Search locations</string>
+    <string name="blaze_campaign_preview_missing_image_dialog_text">Please add an image for the Blaze campaign</string>
+    <string name="blaze_campaign_preview_missing_image_dialog_positive_button">Add Image</string>
+    <string name="blaze_campaign_preview_missing_content_dialog_text">Please add a tagline and a description for your Blaze campaign</string>
+    <string name="blaze_campaign_preview_missing_content_dialog_positive_button">Add</string>
```
Comment (hichamboushaba) [inline WooCommerce/src/main/res/values/strings.xml:3852]: I used just "Add" to keep it short, but please let me know if you have a better suggestion.

**Review by 0nko — CHANGES_REQUESTED** [review]
~~While testing the empty tagline scenario I encountered a crash when I started typing in one of the fields:~~

```
java.lang.IndexOutOfBoundsException: Index: 0, Size: 0
at java.util.ArrayList.set(ArrayList.java:453)
at com.woocommerce.android.ui.blaze.creation.ad.BlazeCampaignCreationEditAdViewModel.updateSuggestion(BlazeCampaignCreationEditAdViewModel.kt:139)
at com.woocommerce.android.ui.blaze.creation.ad.BlazeCampaignCreationEditAdViewModel.onTagLineChanged(BlazeCampaignCreationEditAdViewModel.kt:113)
```

~~It happens because the suggestions don't get loaded so we must check if there are any in `updateSuggestion()`.~~

I thought it wasn't possible to get there without the dialogs but actually this is not caused by this PR. I'll fix it separately.

**Review by 0nko — APPROVED** [review]
The tests work as expected 👍

**Comment by hichamboushaba** [comment]
> I thought it wasn't possible to get there without the dialogs but actually this is not caused by this PR. I'll fix it separately.

Thanks @0nko

**Pull request merged**